libs/remote_storage/src/azure_blob.rs

```diff
@@ -322,6 +323,17 @@ impl RemoteStorage for AzureBlobStorage {
         }
         Ok(())
     }
+
+    #[allow(clippy::diverging_sub_expression)]
+    async fn time_travel_recover(
```

Comment:
Since we will not use object-level operations for the Azure case, I'm not sure we need to try and make the overall interface generic across storage backends.

The scrubber currently doesn't use GenericRemoteStorage at all, so if we wanted object-level recovery functionality to show up as e.g. an extra scrubber subcommand, then we need to choose whether:
A) to make GenericRemoteStorage capable enough for everything we need our offline tools to do
B) or whether to just limit the scope of GenericRemoteStorage to what the pageserver/safekeepers need, and let our offline tools use the cloud APIs directly.

I lean toward B, but I haven't thought about it deeply.

Reply:
I added it to the `remote_storage` crate because I wanted to add the generic restore functionality to the pageserver. potentially also the safekeepers would benefit from it, it should ideally be something that is not pageserver specific.

Note that this is not a per-object API, it's a per-prefix one. You can run it over a single file, over multiple objects in a tenant's prefix, or in the entire bucket if you want. It's a low level operation.

Regarding the question whether the scrubber should use `remote_storage`, I think yes.
Eventually, we will probably want to run the scrubber for all object storages used by us in production, like Azure Blobs, GCP storage, etc.

I'm open to making this a scrubber command though if you still think that a HTTP request is not good. We can always move the code into the `remote_storage` crate.

Reply:
marking as resolved as we talked about it in our 1:1 last week.